exporter.go

```diff
@@ -94,3 +99,126 @@ func (h *httpExporter) htmlHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 }
+
+func newConfluenceExporter(confluenceHost string, confluenceCredentials string, confluencePageID string, client httpClient) (*confluenceExporter, error) {
+	if confluenceHost == "" {
+		return nil, fmt.Errorf("confluenceHost is required")
+	}
+	if confluencePageID == "" {
+		return nil, fmt.Errorf("confluencePageID is required")
+	}
+	return &confluenceExporter{
+		confluenceHost:        confluenceHost,
+		confluenceCredentials: confluenceCredentials,
+		confluencePageID:      confluencePageID,
+		client:                client,
+		mutex:                 sync.Mutex{},
+		abouts:                make(map[string]about)}, nil
+}
+
+type confluenceExporter struct {
+	confluenceHost        string
+	confluenceCredentials string
+	confluencePageID      string
+	client                httpClient
+	mutex                 sync.Mutex //protects abouts
+	abouts                map[string]about
+}
+
+func (h *confluenceExporter) handle(ab about) error {
+	h.mutex.Lock()
+	defer h.mutex.Unlock()
+	h.abouts[ab.Service.Name] = ab
+	a := []about{}
+	for _, about := range h.abouts {
+		a = append(a, about)
+	}
+	var b bytes.Buffer
+	mainTemplate, err := template.ParseFiles(confluenceTemplatePath)
+	if err != nil {
+		return fmt.Errorf("Couldn't find template file for confluence page body: (%v)", err)
+	}
+	if err = mainTemplate.Execute(&b, struct{ Abouts []about }{Abouts: a}); err != nil {
+		return fmt.Errorf("Couldn't render template file for confluence page body: (%v)", err)
+	}
+
+	page, err := h.getConfluencePage(h.confluencePageID)
+	if err != nil {
+		return fmt.Errorf("Could not get confluence page with ID %v: (%v)", h.confluencePageID, err)
+	}
+	page.Body = body{storage{Value: b.String(), Representation: "storage"}}
+	page.Version.Number = page.Version.Number + 1
+
+	if err = h.updateConfluencePage(page); err != nil {
+		return fmt.Errorf("Could not update confluence page with ID %v: (%v)", h.confluencePageID, err)
+	}
+	return nil
+}
+
+func (h *confluenceExporter) getConfluencePage(pageID string) (confluencePage, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/wiki/rest/api/content/%s", h.confluenceHost, h.confluencePageID), nil)
+	if err != nil {
+		return confluencePage{}, fmt.Errorf("Could not get response from %v: (%v)", req.URL.String(), err)
```

Comment:
Error message is not appropriate? Should read "Could not create request ..." right?